CMakePresets.json

```diff
@@ -1,81 +1,320 @@
+        "VCPKG_HOST_TRIPLET": "x64-windows"
+      },
+      "toolset": {
+        "value": "host=x64",
+        "strategy": "external"
+      },
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      },
+      "hidden": true
+    },
+    {
+      "name": "clang-cl",
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "clang-cl.exe"
+      },
+      "toolset": {
+        "value": "host=x64",
+        "strategy": "external"
+      },
+      "condition": {
+        "type": "equals",
+        "lhs": "${hostSystemName}",
+        "rhs": "Windows"
+      },
+      "hidden": true
+    },
+    {
+      "name": "vcpkg",
+      "cacheVariables": {
+        "CMAKE_TOOLCHAIN_FILE": {
+          "value": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
+          "type": "FILEPATH"
+        }
+      },
+      "hidden": true
+    },
+    {
+      "name": "windows-x64-release",
+      "inherits": [ "base", "x64", "msvc", "release", "vcpkg" ],
+      "hidden": false,
+      "description": "Windows x64 Release",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
         "VCPKG_TARGET_TRIPLET": "x64-windows-static-md"
       }
     },
     {
-      "name": "debug",
-      "inherits": "default",
-      "binaryDir": "${sourceDir}/build/${presetName}"
+      "name": "windows-x64-debug",
+      "inherits": [ "base", "x64", "msvc", "vcpkg" ],
+      "hidden": false,
+      "description": "Windows x64 Debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "VCPKG_TARGET_TRIPLET": "x64-windows-static-md"
+      }
     },
     {
-      "name": "release",
-      "inherits": "default",
-      "binaryDir": "${sourceDir}/build/${presetName}",
+      "name": "windows-arm64-release",
+      "inherits": [ "base", "arm64", "msvc", "release", "vcpkg" ],
+      "hidden": false,
+      "description": "Windows ARM64 Release",
       "cacheVariables": {
         "CMAKE_BUILD_TYPE": "Release",
-        "CMAKE_OPTIMIZE_DEPENDENCIES": "ON",
-        "CMAKE_INTERPROCEDURAL_OPTIMIZATION": "ON",
-        "CMAKE_UNITY_BUILD": "ON",
-        "CMAKE_VISIBILITY_INLINES_HIDDEN": "ON",
-        "CMAKE_CXX_VISIBILITY_PRESET": "hidden",
-        "CMAKE_TRY_COMPILE_CONFIGURATION": "Release"
+        "VCPKG_TARGET_TRIPLET": "arm64-windows-static-md"
+      }
+    },
+    {
+      "name": "windows-arm64-debug",
+      "inherits": [ "base", "arm64", "msvc", "vcpkg" ],
+      "hidden": false,
+      "description": "Windows ARM64 Debug",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "VCPKG_TARGET_TRIPLET": "arm64-windows-static-md"
+      }
+    },
+    {
+      "name": "linux-x64-release",
+      "inherits": [ "base", "vcpkg", "release" ],
+      "hidden": false,
+      "description": "Linux x64 Release with GCC",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_C_COMPILER": "gcc",
+        "CMAKE_CXX_COMPILER": "g++"
```

Comment:
Can we _not_ specify compiler executables like this, please? Let's just keep things simple and reference `CMAKE_TOOLCHAIN_FILE` presets like we do in DOSBox Staging. I'd like these presets to be as minimal as possible, let's not overdo it.

https://github.com/dosbox-staging/dosbox-staging/blob/main/CMakePresets.json